backend/crawlers/ddc.js

```diff
@@ -0,0 +1,167 @@
+  const storageClusters = clusters.filter((c) => !JSON.parse(c.params).type || JSON.parse(c.params).type === 'storage');
+
+  const storageFreeResource = _.sum(storageNodes.map((n) => n.node.free_resource));
+  const storageUsedResource = _.sum(storageClusters
+    .map((c) => c.cluster.resource_per_vnode * c.cluster.vnodes.length));
+
+  const storageCapacity = storageFreeResource + storageUsedResource;
+  const storageCapacityMetricId = await insertMetricAndReturnId(dbClient, 'storageCapacity');
+  await insertMetricValue(dbClient, storageCapacityMetricId, storageCapacity);
+
+  const avgPricePerStorage = _.mean(storageClusters
+    .map((c) => c.cluster.total_rent / (c.cluster.resource_per_vnode * c.cluster.vnodes.length)));
+  const avgPricePerStorageMetricId = await insertMetricAndReturnId(dbClient, 'avgPricePerStorage');
+  await insertMetricValue(dbClient, avgPricePerStorageMetricId, avgPricePerStorage);
+
+  const buckets = await ddcBucketQuery.bucketList(contract);
+
+  const uniqueAccounts = new Set(buckets.map((b) => b.bucket.owner_id));
+  const uniqueAccountsMetricId = await insertMetricAndReturnId(dbClient, 'uniqueAccounts');
+  await insertMetricValue(dbClient, uniqueAccountsMetricId, uniqueAccounts.size);
+
+  return { storageNodes, gatewayNodes };
+}
+
+async function collectStorageNodeMetrics(dbClient, node) {
+  const { url } = JSON.parse(node.params);
+  if (!url) {
+    logger.info(`Can't collect storage node ${node.node_id} metrics (url undefined)`);
+    return;
+  }
+
+  let metrics;
+  try {
+    const response = await axios.get(`${url}/metrics/stats`);
+    metrics = response.data;
+  } catch (error) {
+    logger.error(loggerOptions, `Can't collect storage node metrics ${node.node_id} on ${url} (${error.toString()})`);
+    return;
+  }
+  const dataStoredMetricId = await insertMetricAndReturnId(dbClient, 'dataStoredBytes', node.node_id);
+  const piecesStoredMetricId = await insertMetricAndReturnId(dbClient, 'piecesStored', node.node_id);
+  await insertMetricValue(dbClient, dataStoredMetricId, metrics.dataStoredBytes);
+  await insertMetricValue(dbClient, piecesStoredMetricId, metrics.piecesStored);
+}
+
+async function collectStorageMetrics(dbClient, nodes) {
+  logger.info(loggerOptions, 'Collecting storage metrics...');
+  return Promise.all(nodes.map((n) => collectStorageNodeMetrics(dbClient, n)));
+}
+
+async function collectGatewayNodeMetrics(dbClient, node) {
+  const { url } = JSON.parse(node.params);
+  if (!url) {
+    logger.info(`Can't collect gateway node ${node.node_id} metrics (url undefined)`);
+    return;
+  }
+
+  let metrics;
+  try {
+    const response = await axios.get(`${url}/metrics/stats`);
+    metrics = response.data;
+  } catch (error) {
+    logger.error(loggerOptions, `Can't collect gateway node metrics ${node.node_id} (${JSON.stringify(error)})`);
+    return;
+  }
+
+  const piecesViewedMetricId = await insertMetricAndReturnId(dbClient, 'piecesViewed', node.node_id);
```

Comment:
What about adding these metrics to an array and using a loop here? With such approach it will be much easier to add new metrics

Reply:
done